### PR TITLE
Add Sequelize ORM and Replace SQL Queries

### PR DESCRIPTION
**Pull request by chnthkksn**
In this PR, I have made the following modifications:

1. Added Sequelize as a dependency
2. Configured Sequelize connection
3. Migrated SQL queries to Sequelize
4. Created database models
5. Refactored code for Sequelize usage

### CODE REVIEW

**Comment by chnthkksn** [comment]
There were some conflicts with the main branch. I have resolved them.

File: server/models/categoryModel.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const { DataTypes } = require("sequelize");
+const sequelize = require("../database/sequelize");
+
+const Category = sequelize.define(
+  "categories",
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      allowNull: false,
```
Comment (nimi-kaushal) [inline server/models/categoryModel.js]: remove : allowNull: false,

File: server/models/transactionModel.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const { DataTypes } = require("sequelize");
+const sequelize = require("../database/sequelize");
+
+const Transaction = sequelize.define(
+  "transaction",
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      allowNull: false,
```
Comment (nimi-kaushal) [inline server/models/transactionModel.js]: remove    `allowNull: false,`

File: server/routes/budgetRouter.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-const router = require('express').Router();
```
Comment (nimi-kaushal) [inline server/routes/budgetRouter.js:1]: use single ' quotation intsend of  " double quotation

File: server/models/transactionModel.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    date: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "transactions",
```
Comment (nimi-kaushal) [inline server/models/transactionModel.js]:
```
      sequelize 
      tableName: 'transaction',
      underscored: true
```

File: server/models/budgetModel.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    estimated_budget: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "budgets",
```
Comment (nimi-kaushal) [inline server/models/budgetModel.js]:
```
sequelize 
  tableName: 'budget',
      underscored: true
```

File: server/controllers/budgetController.js [hunk 6]
```diff
@@ -1,74 +1,71 @@
-const Pool = require("../database/connection");
+const budget = require("../models/budgetModel");
+const Transaction = require("../models/transactionModel");
+const Category = require("../models/categoryModel");
 
 module.exports = {
+  getBudget: async (req, res) => {
```
Comment (nimi-kaushal) [inline server/controllers/budgetController.js:6]: assign to `const BudgetContoller `and export the module ,  then no need to import as seperate methods

**Review by nimi-kaushal — CHANGES_REQUESTED** [review]

File: server/models/budgetModel.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+const { DataTypes } = require("sequelize");
+const sequelize = require("../database/sequelize");
+
+const Budget = sequelize.define(
```
Comment (nimi-kaushal) [inline server/models/budgetModel.js]: extend from model

File: server/controllers/budgetController.js (hunk 6 above)
Comment (chnthkksn) [reply server/controllers/budgetController.js:6]: fixed in commit 099eb1dd4c684658b8f51ed33da07bae864438a8

File: server/models/budgetModel.js (hunk 7 above)
Comment (chnthkksn) [reply server/models/budgetModel.js]: removed sequelize.define and added class Model extends in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

File: server/models/budgetModel.js (hunk 5 above)
Comment (chnthkksn) [reply server/models/budgetModel.js]: added in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

File: server/models/transactionModel.js (hunk 2 above)
Comment (chnthkksn) [reply server/models/transactionModel.js]: removed `allowNull: false` for primary keys in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

File: server/models/transactionModel.js (hunk 4 above)
Comment (chnthkksn) [reply server/models/transactionModel.js]: added in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

File: server/models/categoryModel.js (hunk 1 above)
Comment (chnthkksn) [reply server/models/categoryModel.js]: removed `allowNull: false` for primary keys in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

**Review by nimi-kaushal — APPROVED** [review]

**Pull request merged**